src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -0,0 +1,1511 @@
+					}
+					mode_task_state = mode_led_idle_state;
+					led_mode = mode_led_wps_active;
+				}
+			}
+		}
+	}
+}
+
+int wifi_have_input(void)
+{
+	int result;
+	// m2mStub_EintDisable ();
+	result = buffer_count;
+	// m2mStub_EintEnable ();
+	return result;
+}
+
+uint8_t wifi_get_next(void)
+{
+	uint8_t result = 0x00;
+	//
+	// The buffer count is also managed in an ISR so protect this code
+	//
+	m2mStub_EintDisable();
+	if (buffer_count != 0) {
+		result = input_buffer[output_index];
+		output_index = (output_index + 1) % INPUT_BUFFER_SIZE;
+		buffer_count -= 1;
+	}
+	m2mStub_EintEnable();
+	return result;
+}
+
+uint8_t wifi_get_next_to(uint32_t timeout)
+{
+	platform_timeout_s t;
+	uint8_t count = 0;
+	int input_count = 0;
+	platform_timeout_set(&t, timeout);
+
+	do {
+		input_count = wifi_have_input();
+		if (input_count != 0) {
+			break;
+		}
+		//
+		// We must run the platform tasks or incomming data will not be transferred
+		// to the input buffers
+		//
+
+		platform_tasks();
+
+	} while (!platform_timeout_is_expired(&t));
+
+	if (input_count != 0) {
+		count = wifi_get_next();
+	}
+	return count;
+}
+
+void do_gdb_send(void)
+{
+	send(gdb_client_socket, &(gdb_send_queue[gdb_send_queue_out].packet[0]), gdb_send_queue[gdb_send_queue_out].len, 0);
+	m2mStub_EintDisable();
+	gdb_send_queue_out = (gdb_send_queue_out + 1) % SEND_QUEUE_SIZE;
+	gdb_send_queue_length -= 1;
+	m2mStub_EintEnable();
+}
+
+void do_uart_debug_send(void)
+{
+	send(uart_debug_client_socket, &(uart_debug_send_queue[uart_debug_send_queue_out].packet[0]),
+		uart_debug_send_queue[uart_debug_send_queue_out].len, 0);
+	m2mStub_EintDisable();
+	uart_debug_send_queue_out = (uart_debug_send_queue_out + 1) % SEND_QUEUE_SIZE;
+	uart_debug_send_queue_length -= 1;
+	m2mStub_EintEnable();
+}
+
+void do_awo_trace_send(void)
+{
+	send(swo_trace_client_socket, &(swo_trace_send_queue[swo_trace_send_queue_out].packet[0]),
+		swo_trace_send_queue[swo_trace_send_queue_out].len, 0);
+}
+
+void send_uart_data(uint8_t *buffer, uint8_t length)
+{
+	m2mStub_EintDisable();
+	memcpy(uart_debug_send_queue[uart_debug_send_queue_in].packet, buffer, length);
+	uart_debug_send_queue[uart_debug_send_queue_in].len = length;
+	uart_debug_send_queue_in = (uart_debug_send_queue_in + 1) % SEND_QUEUE_SIZE;
+	uart_debug_send_queue_length += 1;
+	m2mStub_EintEnable();
+	do_uart_debug_send();
+}
+
+void send_swo_trace_data(uint8_t *buffer, uint8_t length)
+{
+	bool sendIt = false;
+
+	m2mStub_EintDisable();
+	memcpy(swo_trace_send_queue[swo_trace_send_queue_in].packet, buffer, length);
+	swo_trace_send_queue[swo_trace_send_queue_in].len = length;
+	swo_trace_send_queue_in = (swo_trace_send_queue_in + 1) % SEND_QUEUE_SIZE;
+	swo_trace_send_queue_length += 1;
+	sendIt = swo_trace_send_queue_length == 1 ? true : false;
+
+	m2mStub_EintEnable();
+	if (sendIt) {
+		do_awo_trace_send();
+	}
+}
+
+void wifi_gdb_putchar(uint8_t theChar, int flush)
+{
+	send_buffer[send_count++] = theChar;
+	if (flush != 0) {
```

Comment:
This should probably also test that `send_count` is the size of `send_buffer`?

Reply:
Add size check